core/services/pipeline/orm.go

```diff
@@ -186,6 +186,9 @@ func (o *orm) DeleteRun(id int64) error {
 }
 
 func (o *orm) UpdateTaskRunResult(taskID uuid.UUID, result Result) (run Run, start bool, err error) {
+	if result.OutputDB().Valid && result.ErrorDB().Valid {
+		panic("run result must specify either output or error, not both")
```

Comment:
This would fail DB validation anyway, but is honestly a programming error and should be caught higher up